src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -146,6 +147,44 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 	}
 }
 
+function reorderGridData(source: IDataResource): void {
+	// Get Column Names list from the data resource schema
+	const columnNames: string[] = source.schema.fields.map(field => field.name);
+	// Check to see if data source is ordered properly based on schema
+	// Papermill executed notebooks with KQL for instance will organize the
+	// row data in alphabetical order as a result the outputted grid will be
+	// unordered and not based on the data resource schema
+	if (source.data.length > 0) {
+		let rowKeys = Object.keys(source.data[0]);
+		if (!equals(columnNames, rowKeys)) {
+			// SQL notebooks do not use columnName as key (instead use indices)
+			// Indicies indicate the row is ordered properly
+			// We must check the data to know if it is in index form
+			let isValidOrder = false;
```

Comment:
Don't really think this is a good name since it's not really a "valid" order. I would suggest `isNotIndexOrderKeys` or something similar.